src/capi_frontend/capi.cpp

```diff
@@ -753,6 +753,69 @@ OVMS_Status* OVMS_Inference(OVMS_Server* serverPtr, OVMS_InferenceRequest* reque
     return nullptr;
 }
 
+OVMS_Status* OVMS_GetServableState(OVMS_Server* serverPtr, const char* servableName, int64_t servableVersion, OVMS_ServableState* state) {
+    if (serverPtr == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "server"));
+    }
+    if (servableName == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "servable name"));
+    }
+    if (state == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "servable status"));
+    }
+    //
+    // TODO check inputs
+    // TODO metrics
+    std::unique_ptr<ModelInstanceUnloadGuard> modelInstanceUnloadGuard;
+    std::unique_ptr<ovms::Pipeline> pipelinePtr;
+    ovms::Server& server = *reinterpret_cast<ovms::Server*>(serverPtr);
+    ModelManager* modelManager{nullptr};
+    auto status = getModelManager(server, &modelManager);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    std::shared_ptr<ovms::ModelInstance> modelInstance = modelManager->findModelInstance(servableName, servableVersion);
+
+    if (modelInstance == nullptr) {
+        SPDLOG_DEBUG("Requested model: {} does not exist. Searching for pipeline with that name...", servableName);
+        PipelineDefinition* pipelineDefinition = nullptr;
+        std::unique_ptr<PipelineDefinitionUnloadGuard> unloadGuard;
+        pipelineDefinition = modelManager->getPipelineFactory().findDefinitionByName(servableName);
+        if (!status.ok() || !pipelineDefinition) {
+            return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::MODEL_NAME_MISSING));
+        }
+
+        switch (pipelineDefinition->getStateCode()) {
+        case ovms::PipelineDefinitionStateCode::BEGIN:
+            *state = OVMS_ServableState::OVMS_BEGIN;
+            break;
+        case ovms::PipelineDefinitionStateCode::RELOADING:
+        case ovms::PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED:
+        case ovms::PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED_REQUIRED_REVALIDATION:
+            *state = OVMS_ServableState::OVMS_LOADING;
+            break;
+        case ovms::PipelineDefinitionStateCode::AVAILABLE:
+        case ovms::PipelineDefinitionStateCode::AVAILABLE_REQUIRED_REVALIDATION:
+            *state = OVMS_ServableState::OVMS_AVAILABLE;
+            break;
+        case ovms::PipelineDefinitionStateCode::RETIRED:
+            *state = OVMS_ServableState::OVMS_RETIRED;
+            break;
+        }
```

Comment:
Encapsulate this in conversion function and test all the conversions.
Do the same for model instance states. You can already also add the handling for MP, as it uses the same states as DAGs.

Reply:
Then you won't need to mock the DAG to change states.
Also add 1 integration test to verify:
1) OVMS_GetServableState before even OVMS is started - should get error
2) ovms is started but did not yead read config.json - should get error
3) wait for server to be ready but with empty config (no model/dag/mediapipe graph)
4) verify that for each servable we get error
5) change config so that we have model/dag/mediapipe that will load properly
6) verify that for each servable we get available